tests/OpenTK.Tests.Math/QuaternionTests.cs

```diff
@@ -0,0 +1,214 @@
+﻿using Xunit;
+using System;
+using System.Collections.Generic;
+
+namespace OpenTK.Tests.Math
+{
+	/// <summary>
+	/// Generates Quaternion test data.
+	/// </summary>
+	public class QuaternionTestDataGenerator
```

Comment:
Break this out into its own file, preferrably under a relevantly named subdirectory. Perhaps "Generators".

Reply:
What about "TestDataProvider" or "DataProvider"?